include/sdf/Filesystem.hh

```diff
@@ -128,6 +128,14 @@ namespace sdf
       IGN_UTILS_UNIQUE_IMPL_PTR(dataPtr)
     };
   }
+
+  /// \brief The source path replacement if it was parsed from a string,
+  /// instead of a file.
+  const std::string sdfStringSource = "<data-string>";
```

Comment:
Is there a way to make this private API for now? Or do you see this as an important part of a user's workflow?

Reply:
Great point, as far as I know there should not be any public usage, I'll look into into getting it into implementation land.

Reply:
https://github.com/osrf/sdformat/pull/537/commits/745ea3bf520de5c16927d2dccade26bd3db73b85

Reply:
I found a place outside of libsdformat where this is used: https://github.com/ignitionrobotics/ign-gazebo/blob/57d583390c6c9d3f39f520f0570a9637f5060fcd/src/Util.cc#L289. I suggest:

1. We revert to `data-string` instead of `<data-string>`. We can put the `< >` just in the console message.
2. We revert to `urdf string` instead of `<urdf-string>`.
3. We explicitly make this part of the public API by creating constants in `Types.hh`.

Reply:
that works! I've also verified that the paths from `TinyXML2` are absolute and will always be prefixed with a `/`, so there should be no chance for the edge case where the file paths names are `data-string` or `urdf string`

Reply:
https://github.com/osrf/sdformat/pull/537/commits/b922496a5ae2dc4acf2c469721678b7952ff8c4d

Reply:
Sorry to page in late here; is it hard to fix those downstream usages in subsequent PR?
First off, (1) and (2) are inconsistent. Second off, it's still kinda defective in terms of #536.

Given both are pre-release, I think we could change?

Lemme take a quick whack at this.

Reply:
Filed #551

Reply:
Filed https://github.com/ignitionrobotics/ign-gazebo/pull/794

Reply:
Not sure why (1) and (2) are inconsistent. I'm just suggesting we revert to the original strings: `data-string` and `urdf string`.

Reply:
Sorry for the lack of visibility, the suggestion to revert back to `data-string` and `urdf string` without the triangle brackets was made during a call. 

https://github.com/osrf/sdformat/pull/537/commits/b922496a5ae2dc4acf2c469721678b7952ff8c4d is the final commit that reverts it and puts together the triangle brackets separately.

Reply:
> Not sure why (1) and (2) are inconsistent. 

Soz, that was poorly worded. I meant to say having one form the file appear with `< >` (console output) and another form without (API) is kinda jarring / inconsistent.

> I'm just suggesting we revert to the original strings [...]

I think this is a poorly motivated solution, tho? (possibly easy to fix downstream rather than have weird "imbalanced" tokens)
Hence the PR + issue.

> Sorry for the lack of visibility [...]

No worries! I think the motivation was sufficiently visible, just didn't agree w/ motivation :sweat_smile: